haskell-style.md

```diff
@@ -660,3 +660,329 @@ module Driver
+  -- veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea
+  -- commodo consequat. Duis aute irure dolor in reprehenderit in voluptate
+  -- velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat
+  -- cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id
+  -- est laborum.
+  theFunction
+
+  -- | The docs
+  --
+  -- Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod
+  -- tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim
+  -- veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea
+  -- commodo consequat.
+  -- Duis aute irure dolor in reprehenderit in voluptate velit esse cillum
+  -- dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non
+  -- proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
+  theFunction
+  ```
+
+  Good
+
+  ```hs
+  -- | The docs
+  --
+  -- Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod
+  -- tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim
+  -- veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea
+  -- commodo consequat.
+  --
+  -- Duis aute irure dolor in reprehenderit in voluptate velit esse cillum
+  -- dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non
+  -- proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
+  --
+  theFunction
+  ```
+
+- Lists receive a hanging indent
+
+  Bad
+
+  ```hs
+  -- | The docs
+  --
+  -- 1. Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod
+  -- tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim
+  -- veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea
+  -- commodo consequat.
+  -- 2. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum
+  -- dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non
+  -- proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
+  --
+  theFunction
+  ```
+
+  Good
+
+  ```hs
+  -- | The docs
+  --
+  -- 1. Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod
+  --    tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim
+  --    veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea
+  --    commodo consequat.
+  -- 2. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum
+  --    dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non
+  --    proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
+  --
+  theFunction
+  ```
+
+### Module organization
+
+- Organize your exports by logical groups or progressive disclosure and use
+  section headings
+
+  Bad
+
+  ```hs
+  module Foo
+    ( getFoo
+    , updateBar
+    , internalDeleteFoo
+    , deleteFoo
+    )
+  ```
+
+  Good
+
+  ```hs
+  module Foo
+    (
+    -- * Foo
```

Comment:
I think some guidance on sections would be useful. I've seen bare nouns as well as sentences here

Reply:
I'm open to guidance or leaving it un(der)-specified. What do you propose?

Reply:
For some packages in the wild, I've seen `Data Types`, `Functions`, which doesn't seem useful to me. I've also seen `Tutorial`, `Exports`, `<whatever the module's actually about>`, `Internal`. I dunno, I'm fine to let people improvise here until we have a better idea.

Reply:
> I dunno, I'm fine to let people improvise here until we have a better idea.

That's my take. It's unclear if these are Summary (which should then be declarative answers to "This section... {Summary}" or just Titles/Topics/Headings (which is why `Foo` or `Internal` works). Because of this, I was fine leaving this bit open-ended.